### PR TITLE
Add cloudbuild.yaml for hello-app:1.0 & 2.0

### PR DESCRIPTION
**Pull request by NimJay**
This pull-request simply adds a `cloudbuild.yaml` file that tells [Google Cloud Build](https://cloud.google.com/build) to build and push the image `hello-app:1.0` image at:
* `gcr.io/google-samples/hello-app:1.0`
* `us-docker.pkg.dev/google-samples/containers/gke/hello-app:1.0`
  
Some concerns:

1. I have yet to create the associated Cloud Build trigger. We still need to decide on which GCP project to use for the Cloud Build triggers. @donmccasland mentioned that we will be using the `google-samples` projects, but @askmeegs mentioned that it might be better to use a different project.

2. The comment at the top of the cloudbuild.yaml file might be redundant but I've added it in case the users of our GKE tutorials are confused by the file.

3. This commit only builds hello-app:1.0. It does not build hello-app:2.0. @askmeegs pointed out that we will need to update the [main.go file](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/master/hello-app/main.go#L48) (i.e., replace `"Version: 1.0.0\n"`) and _then_ build and push hello-app:2.0. I am currently looking at how we can [pass data between Cloud Build steps](https://medium.com/google-cloud/how-to-pass-data-between-cloud-build-steps-de5c9ebc4cdd) to see if I can replace the version number via Cloud Build itself. I may need to consult Dina on this. 

4. I've used `us-docker.pkg.dev/google-samples/containers/gke/hello-app:1.0` for the Artifact Registry URL. I chose this URL because it had already been created (in [Oct 2020](https://pantheon.corp.google.com/artifacts/docker/google-samples/us/containers/gke%2Fhello-app?project=google-samples)).

### CODE REVIEW

**Comment by askmeegs** [comment]
1. I am pro using a new, dedicated GKE samples CI project, solely owned by our team, because `google-samples` is used by other teams and I believe this project will result in several triggers, and possibly some IAM changes, and I worry about other folks, not on our team, accidentally deleting these resources. (and us, accidentally clobbering theirs.)
2. comment is fine!
3. updating the main.go in the cloudbuild.yaml inline is probably the cleanest way to do this (and avoids duplicating the code) but I'm not sure if it's an anti-pattern.
4. yes, that AR registry is the right spot! `us-docker.pkg.dev/google-samples/containers/gke/<sample-name>:<tag>`

**Comment by askmeegs** [comment]
Tested in my own repo + project. Build passed: 
Screen Shot 2021-06-03 at 10 20 06 AM

GCR:
Screen Shot 2021-06-03 at 10 19 31 AM

AR:
Screen Shot 2021-06-03 at 10 19 59 AM

**Review by askmeegs — APPROVED** [review]

**Comment by askmeegs** [comment]
one small thing, could you add this license header to the top of cloudbuild.yaml ? thanks! https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/master/hello-app/manifests/helloweb-deployment.yaml#L1

**Comment by NimJay** [comment]
Sweet! Thanks, @askmeegs!
I've added the Apache 2.0 license header. 👍
I'll wait on @donmccasland's review before merging.

**Comment by donmccasland** [comment]
> This pull-request simply adds a `cloudbuild.yaml` file that tells [Google Cloud Build](https://cloud.google.com/build) to build and push the image `hello-app:1.0` image at:
> 
> * `gcr.io/google-samples/hello-app:1.0`
> * `us-docker.pkg.dev/google-samples/containers/gke/hello-app:1.0`
> 
> Some concerns:
> 
> 1. I have yet to create the associated Cloud Build trigger. We still need to decide on which GCP project to use for the Cloud Build triggers. @donmccasland mentioned that we will be using the `google-samples` projects, but @askmeegs mentioned that it might be better to use a different project.

We can land this PR first and then build the triggers for it.  I'd like to keep the cloudbuild triggers in the same project we are delivering containers to.  It makes it easier to find and manager later.

To address the fear that other users of google-samples might erase/change our settings and break our automation, let's first document all of the changes needed clearly in a runbook for managing these releases, and then work towards automating the setup with some Terraform.


> 3. This commit only builds hello-app:1.0. It does not build hello-app:2.0. @askmeegs pointed out that we will need to update the [main.go file](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/master/hello-app/main.go#L48) (i.e., replace `"Version: 1.0.0\n"`) and _then_ build and push hello-app:2.0. I am currently looking at how we can [pass data between Cloud Build steps](https://medium.com/google-cloud/how-to-pass-data-between-cloud-build-steps-de5c9ebc4cdd) to see if I can replace the version number via Cloud Build itself. I may need to consult Dina on this.

Let's go ahead and add the steps for modifying the main.go file and then regenerating the container into the cloudbuild.yaml.


> 4. I've used `us-docker.pkg.dev/google-samples/containers/gke/hello-app:1.0` for the Artifact Registry URL. I chose this URL because it had already been created (in [Oct 2020](https://pantheon.corp.google.com/artifacts/docker/google-samples/us/containers/gke%2Fhello-app?project=google-samples)).

Great.  Remember that we are delivering 4 total images:
${GCR}/hello-app:1.0
${GCR}/hello-app:2.0
${GAR}/hello-app:1.0
${GAR}/hello-app:2.0

**Comment by NimJay** [comment]
I have just added a commit that contains the cloudbuild.yaml steps for building hello-app:2.0.

File: hello-app/cloudbuild.yaml
Comment (donmccasland) [inline hello-app/cloudbuild.yaml:42]: cp -p hello-app/main.go hello-app/main.go.bak && sed "s/Version: 1.0.0/Version: 2.0.0/g" hello-app/main.go.bak > hello-app/main.go

**Pull request merged**
